### PR TITLE
Tiny fixes for native wasm support

### PR DESCRIPTION
**Pull request by kripken**
There is now an almost-working path to running wasm natively (binaryen+spidermonkify.py script), which is allowing me to find a bunch of tiny issues. Corresponds to https://github.com/kripken/emscripten/pull/4618.

### CODE REVIEW

File: src/asm2wasm.h [hunk 1]
```diff
@@ -664,6 +664,10 @@ void Asm2WasmBuilder::processAsm(Ref ast) {
     }, debug, false /* do not validate globally yet */);
   }
 
+  // if we see no function tables in the processing below, then the table still exists and has size 0
+
+  wasm.table.initial = wasm.table.max = 0;
```
Comment (dschuff) [inline src/asm2wasm.h]: do we need to add `wasm.table.exists = true` here?

Comment (kripken) [inline src/asm2wasm.h]: Turns out we already do that later down when we create the import - we mark the table as both imported and existing on consecutive lines. Seems ok that way to me.

Comment (dschuff) [inline src/asm2wasm.h]: Ah i see. in that case is it even necessary to have this line then? This line doesn't cause the table to exist, and the table constructor sets both of the limits to 0 anyway?

Comment (kripken) [inline src/asm2wasm.h]: The constructor sets the max to `kMax` (indicating a growable table with no maximum). So the `initial` isn't strictly needed, just the `max`, but it feels nicer I think to set them both here, clarifying that we start the table at fixed size 0.

**Comment by dschuff** [comment]
(Otherwise lgtm)

**Pull request merged**